compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/expressions/BLangQueryExpr.java

```diff
@@ -46,6 +46,8 @@ public class BLangQueryExpr extends BLangExpression implements QueryExpressionNo
     public boolean isTable = false;
     public boolean isMap = false;
 
+    public BLangStatementExpression stmtExpr;
```

Comment:
Why are we maintaining a statement exp within the query expression?

Reply:
`desugared` flag is set to true for `stmtExpr` because it is the return node for query desugar. So, to check the node is desugared or not we used this attribute. 
https://github.com/ballerina-platform/ballerina-lang/pull/37137/files/2e5255b6e5c23067c0f5eded007d054cacd9b60f#diff-a5961a28254babde0aa44e4c1cb05126b5ee68494baccca1c76879dec267b8aeR8092